### PR TITLE
Adding some gRPC tests around cancellation

### PR DESCRIPTION
**Pull request by oldergod**

### CODE REVIEW

File: wire-grpc-tests/src/test/java/com/squareup/wire/GrpcTest.kt [hunk 1]
```diff
@@ -112,7 +110,7 @@ class GrpcTest {
     }
   }
 
-  @Test
+  @ObsoleteCoroutinesApi @Test
```
Comment (oldergod) [inline wire-grpc-tests/src/test/java/com/squareup/wire/GrpcTest.kt]: I was sick with the warnings

Comment (JakeWharton) [reply wire-grpc-tests/src/test/java/com/squareup/wire/GrpcTest.kt]: Put it on the class so you only do it once

File: wire-grpc-tests/src/test/java/com/squareup/wire/GrpcTest.kt [hunk 2]
```diff
@@ -254,20 +250,100 @@ class GrpcTest {
...
+    mockService.enqueue(
+        ReceiveMessage(RouteGuideProto.RouteNote.newBuilder().setMessage("marco").build()))
+    mockService.enqueue(ReceiveComplete)
+
+    val (requestChannel, responseChannel) = routeGuideService.RouteChat()
+    runBlocking {
+      requestChannel.send(RouteNote(message = "marco"))
+      requestChannel.close()
+      // We wait for mockService to process our actions.
+      delay(500)
```
Comment (Egorand) [inline wire-grpc-tests/src/test/java/com/squareup/wire/GrpcTest.kt]: I wonder if we can avoid this kind of delays and make the mock service suspend until it's processed the actions, is that possible?

File: wire-grpc-tests/src/test/java/com/squareup/wire/GrpcTest.kt [hunk 3]
```diff
@@ -49,6 +47,8 @@ import java.time.Duration
 import java.util.concurrent.TimeUnit
 import java.util.concurrent.atomic.AtomicReference
 
+@ExperimentalCoroutinesApi
+@ObsoleteCoroutinesApi
```
Comment (swankjesse) [inline wire-grpc-tests/src/test/java/com/squareup/wire/GrpcTest.kt]: this is out of hand, JetBrains has gone too far

File: wire-grpc-tests/src/test/java/com/squareup/wire/GrpcTest.kt (hunk 2 above)
Comment (swankjesse) [reply wire-grpc-tests/src/test/java/com/squareup/wire/GrpcTest.kt]: possibly? I find interleaving different parts of the code pretty difficult to do neatly

**Comment by swankjesse** [comment]
Good stuff.

File: wire-grpc-tests/src/test/java/com/squareup/wire/GrpcTest.kt (hunk 3 above)
Comment (oldergod) [reply wire-grpc-tests/src/test/java/com/squareup/wire/GrpcTest.kt]: And the thing supposed to replace it isn't released yet...

**Pull request merged**